Neos.Flow/Classes/Http/Component/SecurityEntryPointComponent.php

```diff
@@ -34,46 +34,37 @@ class SecurityEntryPointComponent implements ComponentInterface
+            return $token->getAuthenticationEntryPoint() !== null;
+        });
+
+        if ($tokensWithEntryPoint === []) {
             $this->securityLogger->notice('No authentication entry point found for active tokens, therefore cannot authenticate or redirect to authentication automatically.');
             throw $authenticationException;
         }
 
-        $entryPoint = $firstTokenWithEntryPoint->getAuthenticationEntryPoint();
-        $this->securityLogger->info(sprintf('Starting authentication with entry point of type "%s"', get_class($entryPoint)), LogEnvironment::fromMethodName(__METHOD__));
-
-        // TODO: We should only prevent storage of intercepted request in the session here, but we don't have a different storage mechanism yet.
-        if (!$firstTokenWithEntryPoint instanceof SessionlessTokenInterface && $componentContext->getHttpRequest()->getMethod() === 'GET') {
-            $this->securityContext->setInterceptedRequest($actionRequest->getMainRequest());
-        }
-
-
-        /** @var ResponseInterface $response */
-        $response = $entryPoint->startAuthentication($componentContext->getHttpRequest(), $componentContext->getHttpResponse());
-        $componentContext->replaceHttpResponse($response);
-    }
+        $response = $componentContext->getHttpResponse();
+        foreach ($tokensWithEntryPoint as $token) {
+            $entryPoint = $token->getAuthenticationEntryPoint();
+            $this->securityLogger->info(sprintf('Starting authentication with entry point of type "%s"', \get_class($entryPoint)), LogEnvironment::fromMethodName(__METHOD__));
 
-    /**
-     * Returns the first authenticated token that has an Authentication Entry Point configured, or NULL if none exists
-     *
-     * @return TokenInterface|null
-     */
-    private function getFirstTokenWithEntryPoint(): ?TokenInterface
-    {
-        foreach ($this->securityContext->getAuthenticationTokens() as $token) {
-            if ($token->getAuthenticationEntryPoint() !== null) {
-                return $token;
+            // Only store the intercepted request if it is a GET request (otherwise it can't be resumed properly)
+            // We also don't store the request for "sessionless authentications" because that would implicitly start a session
+            // TODO: Adjust when a session-independent storing mechanism is possible (see https://github.com/neos/flow-development-collection/issues/1693)
+            if (!$token instanceof SessionlessTokenInterface && $componentContext->getHttpRequest()->getMethod() === 'GET') {
+                $actionRequest = $componentContext->getParameter(DispatchComponent::class, 'actionRequest');
+                $this->securityContext->setInterceptedRequest($actionRequest->getMainRequest());
             }
+            $response = $entryPoint->startAuthentication($componentContext->getHttpRequest(), $componentContext->getHttpResponse());
```

Comment:
Something to look at again I guess. I never realised it was done this way. At least by reusing the context response we prevent mixing infos from multiple entry points, but I guess at some point making this a bit more deterministic would be neat.

Reply:
Oh, hang on, I messed up here.. This should of course take the (potentially new) `$response` or the last one will win
```suggestion
            $response = $entryPoint->startAuthentication($componentContext->getHttpRequest(), $response);
```

Reply:
@kitsunet right?